libqtile/widget/base.py

```diff
@@ -29,30 +29,38 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 
-from .. import command, bar, configurable, drawer
+from .. import command, bar, configurable, drawer, confreader
 import six
 import subprocess
 import logging
 import threading
 import warnings
 
+ORIENTATION_NONE = 0
+ORIENTATION_HORIZONTAL = 1
+ORIENTATION_VERTICAL = 2
+ORIENTATION_BOTH = 3
+
 
 class _Widget(command.CommandObject, configurable.Configurable):
     """
-        If width is set to the special value bar.STRETCH, the bar itself
-        will set the width to the maximum remaining space, after all other
+        If length is set to the special value bar.STRETCH, the bar itself
+        will set the length to the maximum remaining space, after all other
         widgets have been configured. Only ONE widget per bar can have the
-        bar.STRETCH width set.
+        bar.STRETCH length set.
 
-        The offset attribute is set by the Bar after all widgets have been
-        configured.
+        The offsetx and offsety attributes are set by the Bar after all widgets
+        have been configured.
     """
-    offset = None
+    offsetx = None
+    offsety = None
     defaults = [("background", None, "Widget background color")]
 
-    def __init__(self, width, **config):
+    def __init__(self, length, orientation, **config):
```

Comment:
Is it possible to use our configuration framework here for both length and orientation? That way they will automatically get picked up by the doc generation code.